bitcoin/Cargo.toml

```diff
@@ -42,6 +42,7 @@ bitcoinconsensus = { version = "0.106.0+26", default-features = false, optional
 actual-serde = { package = "serde", version = "1.0.103", default-features = false, features = [ "derive", "alloc" ], optional = true }
 
 [dev-dependencies]
+internals = { package = "bitcoin-internals", version = "0.3.0", features = ["alloc", "test-serde"] }
```

Comment:
Nit: `alloc` is already in normal dependency, it shouldn't be needed here.

Reply:
Will hold off on ACKing/merging til these comments are addressed becaused it'd be nice to reduce the diff and the amount of noise in Cargo.toml

Reply:
I back-n-forwarded when adding this, you are correct that its not needed but I thought it being identical to the non-dev-dependency with the addition or `test-serde` was more clear. I've been wrong before in manifest files being too explicit so I'll go with the more terse version without giving it much more thought.

Reply:
I personally found it confusing. "What, alloc? Don't we unconditionally depend on it anyway? Oh, we do, it's just redundant" It also makes *this commit* more confusing because of "I don't remember `test-serde` requiring `alloc` and if it does it should be in internals. Well it actually needs `serde_json/alloc` (though that crate probably isn't no-alloc) and `bincode/alloc` but those are enabled by default. (which is OK)